### DISCUSSION TITLE
ISR on-demand need reload page for render actual data on mobile

### DISCUSSION THREAD

**Discussion by eposha** [vercel/vercel#8277] [Help]
Using ISR on-demand on mobile after client routing (Link from next/link or router.push) to revalidated  page I see old data, but if I make reload on this page or go to another page then back to revalidated page again I see actual data. On desktop (local and production with Vercel) all is working correct

What could be the difference of working ISR in a mobile and desktop browsers ?

**Comment by eposha** [comment]
> Using ISR on-demand on mobile after client routing (Link from next/link or router.push) to revalidated page I see old data, but if I make reload on this page or go to another page then back to revalidated page again I see actual data. On desktop (local and production with Vercel) all is working correct
> 
> What could be the difference of working ISR in a mobile and desktop browsers ?

After further investigation, I noticed the same behavior on the desktop version only it happens less often. I also noticed that it does not always happen

**Comment by amyegan** [comment]
Thanks for reaching out @eposha! I believe what you're seeing is normal ISR behavior (not on-demand). Are you able to share your repo or a minimal reproduction? That would help me get a better understanding of what is happening.
